Block/Js.php

```diff
@@ -388,4 +384,30 @@ public function shouldDisableBoltCheckout()
     {
         return !$this->isEnabled() || $this->isPageRestricted() || $this->isIPRestricted();
     }
+
+    /**
+     * Return true if we are on cart page or checkout page
+     */
+    public function IsCartOrCheckoutPage() {
```

Comment:
Regarding of $this->getPageWhitelist() this can include other pages, not only cart or checkout. Perhaps changing the method name into something more generic, isPageAllowed or something similar.

Reply:
isOnPageFromWhiteList